pandas/core/generic.py

```diff
@@ -1915,9 +1915,9 @@ def _repr_data_resource_(self):
     _shared_docs[
         "to_excel"
     ] = """
-    Write %(klass)s to an Excel sheet.
+    Write {klass} to an Excel sheet.
```

Comment:
I think it'd be better if instead of having this defined as a variable `_shared_docs["to_excel"]`, this was the normal docstring of the `to_excel` method.

Then, when reusing it in the `Styler` object, instead of `@doc(_shared_docs["to_excel"], ...)` you'd use `@doc(NDFrame.to_excel, ...)`.

This makes the code more readable, we've been moving in that direction in other docstrings too.

Reply:
and done